server.go

```diff
@@ -777,6 +777,19 @@ func (s *Server) processUnaryRPC(t transport.ServerTransport, stream *transport.
 		if err := s.sendResponse(t, stream, reply, s.opts.cp, opts); err != nil {
 			// TODO: Translate error into a status.Status error if necessary?
 			// TODO: Write status when appropriate.
+			s, ok := status.FromError(err)
+			if !ok {
+				// TODO: Parse possible non-status error
+			} else {
+				switch s.Code() {
+				case codes.InvalidArgument:
```

Comment:
This is probably too specific. It should be OK to do something similar to the error handling on recvMsg: 
```go
	if err == io.EOF {
		// The entire stream is done (for unary RPC only).
		return err
	}

	if st, ok := status.FromError(err); ok {
		if e := t.WriteStatus(stream, st); e != nil {
			grpclog.Printf("grpc: Server.processUnaryRPC failed to write status %v", e)
		}
	} else {
		switch st := err.(type) {
		case transport.ConnectionError:
			// Nothing to do here.
		case transport.StreamError:
			if e := t.WriteStatus(stream, status.New(st.Code, st.Desc)); e != nil {
				grpclog.Printf("grpc: Server.processUnaryRPC failed to write status %v", e)
			}
		default:
			panic(fmt.Sprintf("grpc: Unexpected error (%T) from recvMsg: %v", st, st))
		}
	}
	return err

```